src/core/geometry/fuzzyfactory/FuzzyFactory3.js

```diff
@@ -1,22 +1,24 @@
-const { EPS } = require('../constants')
-const Polygon = require('./poly3')
-const FuzzyFactory = require('./FuzzyFactory')
+const {EPS} = require('../../constants')
 
-const polygonSHaredHash = (shared) => {
-  if (!shared.color) return 'null'
-  return shared.color.join('/')
-}
+const vec3 = require('../../math/vec3')
+
+const poly3 = require('../poly3')
+
+const FuzzyFactory = require('./FuzzyFactory')
 
 // ////////////////////////////////////
-const FuzzyShape3Factory = function () {
+const FuzzyFactory3 = function () {
   this.vertexfactory = new FuzzyFactory(3, EPS)
   this.planefactory = new FuzzyFactory(4, EPS)
   this.polygonsharedfactory = {}
 }
 
-FuzzyShape3Factory.prototype = {
+FuzzyFactory3.prototype = {
   getPolygonShared: function (sourceshared) {
-    let hash = polygonSHaredHash(sourceshared)
+    let hash = 'null'
+    if (sourceshared && (color in sourceshare)) {
```

Comment:
Having trouble seeing how this can work.

Where does color come from?
If sourceshared is falsey, does it share universally with 'null'.
Is 'null' ambiguous (perhaps with a 'null' color).

Suggest something closer to let hash = (sourceshare.color || []).join('/'); for now.

May also find that the following approach is a bit simpler:

let cached = this.polygonsharedfactory[hash];
if (cached === undefined) {
  cached = sourceshared;
  this.polygonsharedfactory[hash] = cached;
}
return cached;

Reply:
Added a comment. Sadly, we haven't decided if color is built into poly3, or poly3 has a property list. TBD